[32mINFO    [0m integration.ha_tests.test_self_healing:test_self_healing.py:486 checking whether the connectivity to the database is working
[32mINFO    [0m integration.ha_tests.test_self_healing:test_self_healing.py:491 Cutting network for postgresql/2
[32mINFO    [0m integration.ha_tests.test_self_healing:test_self_healing.py:497 checking for no connectivity between postgresql/2 and postgresql/3
[32mINFO    [0m integration.ha_tests.test_self_healing:test_self_healing.py:497 checking for no connectivity between postgresql/2 and postgresql/1
[32mINFO    [0m integration.ha_tests.test_self_healing:test_self_healing.py:504 checking for no connectivity between postgresql/2 and the controller
[32mINFO    [0m integration.ha_tests.test_self_healing:test_self_healing.py:511 checking whether the connectivity to the database is not working
[32mINFO    [0m integration.ha_tests.test_self_healing:test_self_healing.py:517 checking whether writes are increasing
[32mINFO    [0m integration.ha_tests.test_self_healing:test_self_healing.py:520 checking whether a new primary was elected
[32mINFO    [0m integration.ha_tests.test_self_healing:test_self_healing.py:527 Restoring network for postgresql/2
[32mINFO    [0m integration.ha_tests.test_self_healing:test_self_healing.py:531 waiting for cluster to become idle
[32mINFO    [0m juju.model:model.py:2958 Waiting for model:
  postgresql/1 [idle] active: Primary
  postgresql/2 [idle] active: 
  postgresql/3 [idle] active: 
[32mINFO    [0m juju.model:model.py:2958 Waiting for model:
  postgresql/2 [idle] active: Primary
[32mINFO    [0m juju.model:model.py:2958 Waiting for model:
  postgresql/3 [idle] active: 
[32mINFO    [0m integration.ha_tests.test_self_healing:test_self_healing.py:536 waiting for the database service to be ready on postgresql/2
[32mINFO    [0m integration.ha_tests.test_self_healing:test_self_healing.py:540 checking whether the connectivity to the database is working
[32mINFO    [0m integration.ha_tests.helpers:helpers.py:196 checking that the former primary is now a replica
[32mINFO    [0m integration.ha_tests.helpers:helpers.py:202 checking that all units are part of the same cluster
[32mINFO    [0m integration.ha_tests.helpers:helpers.py:216 checking that no writes to the database were missed after stopping the writes
[32mINFO    [0m integration.ha_tests.helpers:helpers.py:220 checking that the former primary is up to date with the cluster after restarting
[32mINFO    [0m pytest_operator.plugin:plugin.py:862 Model status:

Model  Controller           Cloud/Region         Version  SLA          Timestamp
test   localhost-localhost  localhost/localhost  3.1.8    unsupported  02:33:28Z

App                  Version  Status  Scale  Charm                Channel  Rev  Exposed  Message
postgresql           14.11    active      3  postgresql                      0  no       Primary
postgresql-test-app           active      1  postgresql-test-app  edge     142  no       received database credentials of the first database

Unit                    Workload  Agent  Machine  Public address  Ports     Message
postgresql-test-app/0*  active    idle   3        10.2.47.148               received database credentials of the first database
postgresql/1*           active    idle   1        10.2.47.188     5432/tcp  Primary
postgresql/2            active    idle   2        10.2.47.15      5432/tcp  
postgresql/3            active    idle   4        10.2.47.5       5432/tcp  

Machine  State    Address      Inst id        Base          AZ  Message
1        started  10.2.47.188  juju-cb3185-1  ubuntu@22.04      Running
2        started  10.2.47.15   juju-cb3185-2  ubuntu@22.04      Running
3        started  10.2.47.148  juju-cb3185-3  ubuntu@22.04      Running
4        started  10.2.47.5    juju-cb3185-4  ubuntu@22.04      Running

[32mINFO    [0m pytest_operator.plugin:plugin.py:868 Juju error logs:

machine-1: 01:37:49 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:37:49 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:37:49 ERROR juju.worker.dependency "broker-tracker" manifold worker returned unexpected error: no container types determined
unit-postgresql-1: 01:37:50 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-0: 01:37:50 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-0: 01:37:50 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-0: 01:37:50 ERROR juju.worker.dependency "broker-tracker" manifold worker returned unexpected error: no container types determined
unit-postgresql-0: 01:37:50 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-3: 01:38:02 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-3: 01:38:02 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-3: 01:38:02 ERROR juju.worker.dependency "broker-tracker" manifold worker returned unexpected error: no container types determined
unit-postgresql-test-app-0: 01:38:02 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-2: 01:38:20 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-2: 01:38:20 ERROR juju.worker.dependency "broker-tracker" manifold worker returned unexpected error: no container types determined
machine-2: 01:38:20 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-2: 01:38:20 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-4: 01:54:12 ERROR juju.worker.dependency "broker-tracker" manifold worker returned unexpected error: no container types determined
machine-4: 01:54:12 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-3: 01:54:12 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-postgresql-1: 01:59:57 ERROR unit.postgresql/1.juju-log Failed to list PostgreSQL database users: connection to server at "10.2.47.157", port 5432 failed: timeout expired

unit-postgresql-1: 02:00:08 ERROR unit.postgresql/1.juju-log Failed to list PostgreSQL database users: connection to server at "10.2.47.157", port 5432 failed: timeout expired

unit-postgresql-test-app-0: 02:04:15 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:04:19 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:04:23 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:04:27 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:04:31 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:04:35 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:04:39 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:04:43 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:04:47 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:04:51 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:04:55 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:04:59 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:05:03 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:05:07 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:05:11 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:05:15 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:05:19 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:05:23 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:05:27 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:05:31 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:05:35 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:05:39 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:05:43 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:05:47 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:05:51 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:05:55 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:05:59 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:06:03 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:06:07 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:06:11 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:06:15 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:06:19 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:06:24 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:06:28 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:06:32 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:06:36 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:06:40 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:06:44 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:06:48 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:06:52 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:06:56 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:07:00 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:07:04 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:07:08 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:07:12 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:07:16 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:07:20 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:07:24 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:07:28 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:07:32 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:07:36 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:07:40 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:07:44 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:07:48 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:07:52 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:07:56 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:08:00 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:08:04 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:08:08 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:08:12 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:08:16 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:08:20 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:08:24 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:08:28 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:08:32 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:08:36 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:08:40 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:08:44 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:08:48 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:08:52 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:08:56 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:09:00 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:09:04 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:09:08 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:09:13 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:09:17 ERROR unit.postgresql-test-app/0.juju-log Unable to drop table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 284, in _on_clear_continuous_writes_action
    cursor.execute("DROP TABLE IF EXISTS continuous_writes;")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:09:19 ERROR unit.postgresql-test-app/0.juju-log Unable to create table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 312, in _on_start_continuous_writes_action
    cursor.execute("CREATE TABLE IF NOT EXISTS continuous_writes(number INTEGER);")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute CREATE TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:09:23 ERROR unit.postgresql-test-app/0.juju-log Unable to create table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 312, in _on_start_continuous_writes_action
    cursor.execute("CREATE TABLE IF NOT EXISTS continuous_writes(number INTEGER);")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute CREATE TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:09:27 ERROR unit.postgresql-test-app/0.juju-log Unable to create table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 312, in _on_start_continuous_writes_action
    cursor.execute("CREATE TABLE IF NOT EXISTS continuous_writes(number INTEGER);")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute CREATE TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:09:31 ERROR unit.postgresql-test-app/0.juju-log Unable to create table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 312, in _on_start_continuous_writes_action
    cursor.execute("CREATE TABLE IF NOT EXISTS continuous_writes(number INTEGER);")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute CREATE TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:09:35 ERROR unit.postgresql-test-app/0.juju-log Unable to create table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 312, in _on_start_continuous_writes_action
    cursor.execute("CREATE TABLE IF NOT EXISTS continuous_writes(number INTEGER);")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute CREATE TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:09:39 ERROR unit.postgresql-test-app/0.juju-log Unable to create table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 312, in _on_start_continuous_writes_action
    cursor.execute("CREATE TABLE IF NOT EXISTS continuous_writes(number INTEGER);")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute CREATE TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:09:43 ERROR unit.postgresql-test-app/0.juju-log Unable to create table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 312, in _on_start_continuous_writes_action
    cursor.execute("CREATE TABLE IF NOT EXISTS continuous_writes(number INTEGER);")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute CREATE TABLE in a read-only transaction

unit-postgresql-test-app-0: 02:09:47 ERROR unit.postgresql-test-app/0.juju-log Unable to create table
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/./src/charm.py", line 312, in _on_start_continuous_writes_action
    cursor.execute("CREATE TABLE IF NOT EXISTS continuous_writes(number INTEGER);")
psycopg2.errors.ReadOnlySqlTransaction: cannot execute CREATE TABLE in a read-only transaction

machine-4: 02:19:42 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: api connection broken unexpectedly
machine-4: 02:19:45 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: [db2fc4] "machine-4" cannot open api: unable to connect to API: dial tcp 10.2.47.147:17070: connect: network is unreachable
machine-4: 02:19:49 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: [db2fc4] "machine-4" cannot open api: unable to connect to API: dial tcp 10.2.47.147:17070: connect: network is unreachable
machine-2: 02:30:50 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: api connection broken unexpectedly
machine-2: 02:30:56 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: [db2fc4] "machine-2" cannot open api: unable to connect to API: dial tcp 10.2.47.147:17070: connect: no route to host
machine-2: 02:31:02 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: [db2fc4] "machine-2" cannot open api: unable to connect to API: dial tcp 10.2.47.147:17070: connect: no route to host
machine-2: 02:31:08 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: [db2fc4] "machine-2" cannot open api: unable to connect to API: dial tcp 10.2.47.147:17070: connect: no route to host
machine-2: 02:31:14 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: [db2fc4] "machine-2" cannot open api: unable to connect to API: dial tcp 10.2.47.147:17070: connect: no route to host
machine-2: 02:31:20 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: [db2fc4] "machine-2" cannot open api: unable to connect to API: dial tcp 10.2.47.147:17070: connect: no route to host
machine-2: 02:31:31 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: [db2fc4] "machine-2" cannot open api: unable to connect to API: dial tcp 10.2.47.147:17070: i/o timeout

[32mINFO    [0m pytest_operator.plugin:plugin.py:947 Forgetting model main...